specs/ChainSpec/LightClient.md

```diff
@@ -0,0 +1,86 @@
+2. The set of block producers for the current and next epochs.
+
+Light clients operate by periodically fetching instances of `LightClientBlockView` via particular RPC end-point desdribed [below](#rpc-end-point).
+
+```rust
+pub struct ValidatorStakeView {
+    pub account_id: AccountId,
+    pub public_key: PublicKey,
+    pub stake: Balance,
+}
+
+pub struct LightClientBlockView {
+    pub prev_block_hash: CryptoHash,
+    pub next_block_inner_hash: CryptoHash,
+    pub inner_lite: BlockHeaderInnerLiteView,
+    pub inner_rest_hash: CryptoHash,
+    pub next_bps: Option<Vec<ValidatorStakeView>>,
+    pub approvals_next: Vec<Option<Signature>>,
+    pub approvals_after_next: Vec<Option<Signature>>,
+}
+```
+
+Recall that the hash of the block is
+
+```rust
+sha256(concat(
+    prev_hash,
+    sha256(concat(
+        sha256(borsh(inner_lite)),
+        sha256(borsh(inner_rest))
+    ))
+))
+```
+
+The fields `prev_block_hash`, `next_block_inner_hash` and `inner_rest_hash` are used to reconstruct the hashes of the current and next block in the following way (where `block_view` is an instance of `LightClientBlockView`):
+
+```rust
+current_block_hash = sha256(concat(
+    block_view.prev_block_hash,
+    sha256(concat(
+        sha256(borsh(block_view.inner_lite)),
+        block_view.inner_rest_hash,
+    ))
+))
+
+next_block_hash = sha256(concat(
+    current_block_hash,
+    block_view.next_block_inner_hash
+))
+```
+
+The light client updates its head with the information from `LightClientBlockView` iff:
+
+1. The height of the block is higher than the height of the current head;
+2. The epoch of the block is equal to the `epoch_id` or `next_epoch_id` known for the current head;
+3. If the epoch of the block is equal to the `next_epoch_id` of the head, then `next_bps` is not `None`;
+4. `approvals_next` and `approvals_after_next` contain signatures that check out against the block producers for the epoch of the block (see next section);
+5. The signatures present in both `approvals_next` and `approvals_after_next` correspond to more than 2/3 of the total stake.
+6. If `next_bps` is not none, `sha256(borsh(next_bps))` corresponds to the `next_bp_hash` in `inner_lite`.
+
+## Signature verification
+
+By construction by the time the `LightClientBlockView` is being validated, the block producers set for its epoch is known.
+
+If the `LightClientBlockView` corresponds to some block at height `h`, then the signatures in the `LightClientBlockView::approvals_next` are signatures on endorsements in the block at height `h+1`, and the signatures in the `LightClientBlockView::approvals_after_next` and signatures on endorsements in the block at height `h+2`. To simplify the protocol we require that the next block and the block after next are both in the same epoch as the block that `LightClientBlockView` corresponds to. It is guaranteed that each epoch has at least one final block for which the next two blocks that build on top of it are in the same epoch.
```

Comment:
When it says:
>then the signatures in the `LightClientBlockView::approvals_next` are signatures on endorsements in the block at height `h+1`

Does it mean?
> then the signatures in the `LightClientBlockView::approvals_next` are signatures on endorsements in the block at height `h+1` of the block at height `h`

Reply:
(while this particular sentence is gone, a similar one remains for `approvals_after_next`)

Yes, but endorsement is always on a block with height `h`, so it's a redundant information. If it was a block with a lower height, it would be a `Skip`